### DISCUSSION TITLE
Fedora RPM Packages

### DISCUSSION THREAD

**Discussion by mowestusa** [ezra-bible-app/ezra-bible-app#1075] [General]
I noticed that for your latest release 1.14 you build packages for Fedora 31, 33, 35, 37.

The current Fedora release is 40, and Fedora only supports releases for about 13 months. Which means that Fedora typically is only supporting and releasing patches for the current release and the release before it. The release that is two releases before the current receives patches for 4 weeks after the new release comes out. So currently the only releases of Fedora that are getting patches and bug fixes are:

Fedora 40
Fedora 39
Fedora 38 was End of Life on 5/21/24

I'm guessing that your CI/CD is just creating packages automatically for those previous versions all of which are not supported by Fedora. I will be testing your package for Fedora 37 on Fedora 40 and will report any issues that I encounter, but perhaps you could make a change to your CI/CD pipeline that builds packages only for the latest Fedora when there is a release and this could save some CPU cycles and perhaps clean up the number of files you have in your releases.

Thanks for providing Ezra Bible App for the Fedora distribution. I'm happy to help in anyway possible. I'm an avid user of Fedora, but I'm not a programmer nor am I a package maintainer for Fedora. I mostly help the distro with documentation.

**Comment by tobias-klein** [comment]
Hi @mowestusa!

Thanks for your feedback!
I tested the latest version 1.14 Fedora 37 package on a Fedora 40 VM and it worked.
At some point I also succcessfully tested on Fedora 39.
That is why I did not add a new package yet.
Those packages are all generated based on GitHub Actions CI/CD.

You are right that we may need to clean this up a bit.
I usually judge this based on the download statistics.
The download counts for the older Fedora versions are indeed quite limited (see below).

We may therefore decide to remove 31, 33, 35.

![image](https://github.com/user-attachments/assets/c2c7fb11-ff1e-4514-9386-46db2870ae6d)

**Comment by mowestusa** [comment]
@tobias-klein, I spent some time testing Ezra on my Fedora 40 install on bare metal, and I also have not had any issues. The issues I'm having are all related to figuring out this new interface and setting things up to get to my Bible Study. I will report any bugs or issues that I have on Fedora 40 but so far so good.

**Comment by tobias-klein** [comment]
Thank you for the feedback @mowestusa! Since this issue is more of a Q&A discussion I am going to convert it to a discussion. We may have more specific individual issues for doing the cleanup and/or creating packages for newer versions of Fedora where the version 37 package is not working any longer.

**Comment by entodoays** [comment]
I am hesitant to install Linux packages manually. I suggest you support having a Flatpak or Appimage that would be easier to maintain in the long run and is also easier to install and uninstall without leaving configuration folders and the like behind.

**Reply by tobias-klein** [reply]
> I am hesitant to install Linux packages manually. I suggest you support having a Flatpak or Appimage that would be easier to maintain in the long run and is also easier to install and uninstall without leaving configuration folders and the like behind.

What about Snap? (see https://snapcraft.io/)

**Comment by entodoays** [comment]
Snap is Ubuntu mainly. Flatpak is more distribution agnostic. Appimages as well.

14 Sept 2024 14:46:05 Tobias Klein ***@***.***>:

**Reply by tobias-klein** [reply]
I just read a bit more about these new packaging technologies. I will have a more detailed look at Flatpak. It does not seem trivial to generate a Flatpak ... but manageable. They have a [guide for Electron applications](https://docs.flatpak.org/en/latest/electron.html).

**Reply by entodoays** [reply]
I'm no developer, but there is a tutorial that looks rather simple: https://docs.fedoraproject.org/en-US/flatpak/tutorial/

**Comment by mowestusa** [comment]
I would second the use of Flatpak just because it would be easier to install on the distributions that I use all of which are none Ubuntu. However, I'm also fine with installing the RPM manually on my Fedora systems. It seems as if you have a CI/CD system for making DEBs and RPMs, so I can be content with that if you want to devote your developer time to other improvements. Thanks for your work.

**Reply by rgbprg3** [reply]
Yes I believe Flatpak is the way this should go. By using flatpak, it should work everywhere that has flatpak installed. No need to compile different versions (deb, rpm) etc. It seems a lot of developers are going this route lately.

**Comment by rgbprg3** [comment]
Update -- installed on Fedora 42 currently with no issues.
